### PR TITLE
feat: the original sender of a specific message should be able to edit that message (AP-39)

### PR DESCRIPTION
**Pull request by nickjjang**
AP-39

### CODE REVIEW

**Review by stavares843 — COMMENTED** [review]
neat! 🎉 

i wonder if we should include the following
- it allows to edit the message but doesn't have the 256 limit char

Captura de ecrã 2021-12-16, às 02 39 46

**Comment by stavares843** [comment]
@WanderingHogan do you know if the edit message should have the 0/256 chars as well?

**Review by stavares843 — COMMENTED** [review]
if we try to edit without any chars, just saves to the first char


example message was ''test'' try to add just a space, saves ''t''

should we return an error maybe? cc @WanderingHogan

**Comment by stavares843** [comment]
Captura de ecrã 2021-12-16, às 02 43 31

clicking on the emoji on edit doesn't work, is that to be implemented on other PR?

**Comment by stavares843** [comment]
is there a max limit time to edit a message? @WanderingHogan

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
also wondering if we should allow users editing a message when is a glyph

**Comment by stavares843** [comment]
when trying to edit a message that's not yours

https://user-images.githubusercontent.com/29093946/146299855-3ad3ff4d-c8ed-496c-9c8c-b8fa49299c8c.mp4

we need to hide the edit message there as well, this or other PR

**Comment by WanderingHogan** [comment]
@stavares843 @josephmcg  the reply message should ahve the same limit as the regular message, and we should set that in the config somewhere so if we can change it, but the replies and new messages are updated

**Comment by WanderingHogan** [comment]
@stavares843 
```if we try to edit without any chars, just saves to the first char

example message was ''test'' try to add just a space, saves ''t''

should we return an error maybe? cc @WanderingHogan```

Do you mean you type 'test' then back it out, and if you hit save it saves as 't'? if that is the case, we should not allow saving blank messages and the remnant t is a bug we should investigate

File: components/views/chat/message/Message.html [hunk 1]
```diff
@@ -29,8 +33,21 @@
...
-    <UiChatImage v-else-if="message.type === 'image'" :source="message.payload.url" alt="" />
+    <template v-if="!messageEdit && message.type === 'text'">
+      <VueMarkdown
+        :source="wrapEmoji(message.payload)"
+        :class="`markdown ${containsOnlyEmoji(message.payload) ? 'bigmoji' : ''}`"
+      />
+      <span class="status editing" v-if="message.editingAt">
+        <UiLoadersSpinner spinning />
+      </span>
+      <span class="status edited" v-else-if="message.editedAt">(edited)</span>
```
Comment (WanderingHogan) [inline components/views/chat/message/Message.html:44]: can you add edited to translations

**Comment by stavares843** [comment]
that's correct yes @WanderingHogan

**Review by WanderingHogan — CHANGES_REQUESTED** [review]
code looks good, can you make the changes sara requested when you get a chance

**Comment by stavares843** [comment]
to summarise
- add a limit of 256 when editing, to avoid users bypassing the chars limit when sending a message when editing
-  add the counter 0/256 chars as well
- when trying to edit a message that's not yours, it saves temporarily and then goes again to the original message, don't allow users to edit messages that are not theirs
- don't allow users to edit a message when is a glyph

any questions let me know @josephmcg

**Comment by WanderingHogan** [comment]
Eric said he will set up the char limit in the config file, so we will get his pr in first so that effort wont be duplicated

File: components/views/chat/message/Message.html (hunk 1 above)
Comment (nickjjang) [reply components/views/chat/message/Message.html]: Fixed

**Comment by nickjjang** [comment]
Hi @WanderingHogan @stavares843 
Can I do it within a new ticket because I changed several files related with sending message, so it makes me resolve many conflict with others. 

> to summarise
> 
> * add a limit of 256 when editing, to avoid users bypassing the chars limit when sending a message when editing
> * add the counter 0/256 chars as well
> * when trying to edit a message that's not yours, it saves temporarily and then goes again to the original message, don't allow users to edit messages that are not theirs
> * don't allow users to edit a message when is a glyph
> 
> any questions let me know @josephmcg

**Comment by stavares843** [comment]
Captura de ecrã 2021-12-16, às 18 34 47

this was showing edited properly before, can you check? @nickjjang

**Comment by nickjjang** [comment]
> Captura de ecrã 2021-12-16, às 18 34 47
> 
> this was showing edited properly before, can you check? @nickjjang

I've updated translation locales submodule for it. need to pull locales to do it.

**Comment by stavares843** [comment]
neat, is here 🎉 thanks!

**Comment by nickjjang** [comment]
@stavares843 @WanderingHogan will you approve it and merge so that I can go ahead to improve the edit message?

**Review by WanderingHogan — APPROVED** [review]

**Pull request merged**